File: http-client/src/test/groovy/io/micronaut/http/client/docs/annotation/headers/PetController.java
package io.micronaut.http.client.docs.annotation.headers;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Header;
import io.micronaut.http.client.docs.annotation.Pet;
import io.reactivex.Single;

@Controller("/pets")
public class PetController {

    @Get("/{name}")
    public HttpResponse<Pet> get(String name, @Header("X-Pet-Client") String clientId) {
        Pet pet = new Pet();
        pet.setName(name);
        pet.setAge(Integer.valueOf(clientId));
        return HttpResponse.ok(pet)
                           .header("X-Pet-Client", clientId);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/annotation/headers/PetController.kt
package io.micronaut.docs.annotation.headers

import io.micronaut.docs.annotation.Pet
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Header

@Controller("/pets")
class PetController {

    @Get("/{name}")
    operator fun get(name: String, @Header("X-Pet-Client") clientId: String): HttpResponse<Pet> {
        val pet = Pet()
        pet.name = name
        pet.age = Integer.valueOf(clientId)
        return HttpResponse.ok(pet)
                .header("X-Pet-Client", clientId)
    }
}
